### PR TITLE
Simplify func ConstructVolumeSpec

### PR DESCRIPTION
**Pull request by cwdsuzhou**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Simplify the codes

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

File: pkg/volume/csi/csi_plugin.go [hunk 1]
```diff
@@ -487,15 +487,10 @@ func (p *csiPlugin) ConstructVolumeSpec(volumeName, mountPath string) (*volume.S
 		switch {
 		case mode == ephemeralDriverMode:
```
Comment (jsafrane) [inline pkg/volume/csi/csi_plugin.go]: I don'd like `case` just with one branch, please simplify it even more. And add comments about falling through to persistent mode.

**Review by jsafrane — COMMENTED** [review]
/assign @vladimirvivien 
/unassign

**Comment by cwdsuzhou** [comment]
ping @vladimirvivien  PTAL

File: pkg/volume/csi/csi_plugin.go [hunk 2]
```diff
@@ -482,20 +482,16 @@ func (p *csiPlugin) ConstructVolumeSpec(volumeName, mountPath string) (*volume.S
...
-		default:
-			spec = p.constructPVSourceSpec(volData[volDataKey.specVolID], volData[volDataKey.driverName], volData[volDataKey.volHandle])
-		}
-	} else {
-		spec = p.constructPVSourceSpec(volData[volDataKey.specVolID], volData[volDataKey.driverName], volData[volDataKey.volHandle])
+	// If inlineEnabled is true and mode is ephemeralDriverMode,
+	// use constructVolSourceSpec to construct volume source spec.
+	// If inlineEnabled is false or mode is persistentDriverMode,
+	// use constructPVSourceSpec to construct volume construct pv source spec.
+	if inlineEnabled && driverMode(volData[volDataKey.driverMode]) == ephemeralDriverMode {
```
Comment (vladimirvivien) [inline pkg/volume/csi/csi_plugin.go]: I support simplifying this code. However, I would recommend:
Break up the if as
```
if inlineEnabled {
    if driverMode(...) == ephemeralDriveMode{ ... }
    return spec, nil
}
spec = ...
```

**Review by vladimirvivien — COMMENTED** [review]
Looks good, with minor change suggestion.

File: pkg/volume/csi/csi_plugin.go (hunk 2 above)
Comment (cwdsuzhou) [reply pkg/volume/csi/csi_plugin.go]: thanks, done!

**Comment by cwdsuzhou** [comment]
ping @vladimirvivien 

PTAL

**Comment by cwdsuzhou** [comment]
ping @vladimirvivien

**Pull request merged**